csvGenerator.ipynb

```diff
@@ -0,0 +1,195 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#download files from the collection items\n",
+    "def download_file(folder_path:str,ids:set, citesphere_token:str,):\n",
+    "    path_list = []\n",
+    "    for (file_id, file_name) in list(ids):\n",
+    "        giles_url = f\"https://diging.asu.edu/geco-giles-staging/api/v2/resources/files/{file_id}/content\"\n",
+    "        os.makedirs(folder_path, exist_ok=True)\n",
+    "        filename = os.path.join(folder_path, f\"{file_name}\")\n",
+    "        path_list.append(filename)\n",
+    "        headers = {\n",
+    "            \"Authorization\": f\"Bearer {citesphere_token}\",\n",
+    "            \"Content-Type\": \"application/pdf;charset=UTF-8\"\n",
+    "            }\n",
+    "        response = requests.get(giles_url, headers=headers)\n",
+    "        if response.status_code == 200:\n",
+    "            with open(filename, \"wb\") as file:\n",
+    "                file.write(response.content)\n",
+    "    return path_list  \n",
+    "\n",
+    "#Returns the set of tuples containing, file_ids and file_name for downloading\n",
+    "\n",
+    "\"\"\"\n",
+    "this function returns a set of tuples that are pair of filename and their file ids\n",
+    "set : (tuple(file_id,file_name),...)\n",
+    "files to download in a collection item are located in the following fields:\n",
+    "uploadedFile, extractedText, pages, image, text, ocr, additionalFiles\n",
+    "\"\"\"\n",
+    "\n",
+    "def get_set(items_list:list):\n",
+    "    ids=set()\n",
+    "    for values in items_list:\n",
+    "        #getting the file ids in uploadedFile, extractedText, pages, image, text, ocr, additionalFiles\n",
+    "        ids.add((values[\"uploadedFile\"][\"id\"],values[\"uploadedFile\"][\"filename\"]))\n",
+    "        if values[\"extractedText\"] and values[\"extractedText\"]!= \"None\":\n",
+    "            ids.add((values[\"extractedText\"][\"id\"],values[\"extractedText\"][\"filename\"]))\n",
+    "        if values[\"pages\"] and values[\"pages\"] != \"None\":\n",
+    "            for value in values[\"pages\"]:\n",
+    "                ids.add((value[\"image\"][\"id\"],value[\"image\"][\"filename\"]))\n",
+    "                ids.add((value[\"text\"][\"id\"],value[\"text\"][\"filename\"]))\n",
+    "                ids.add((value[\"ocr\"][\"id\"],value[\"ocr\"][\"filename\"]))\n",
+    "                for file in value[\"additionalFiles\"]:\n",
+    "                    ids.add((file[\"id\"],file[\"filename\"]))\n",
+    "    return ids\n",
+    "   \n",
+    "#Create the CSV file with all the metadata and file path of the downloaded files.\n",
+    "\n",
+    "\"\"\"\n",
+    "this function writes the meta data to the csv file.\n",
+    "uses a flag to write the fileds at the top of the file in the first time.\n",
+    "\"\"\"\n",
+    "\n",
+    "def write_to_csv(csv_name:str,item:list,flag:int):\n",
+    "    with open(csv_name, 'a', newline='') as file:\n",
```

Comment:
does this mean that the file will be opened and close as often as there are downloaded files? That's not very efficient. It's needs to be opened once at the beginning.